### DISCUSSION TITLE
Tasmota Power Monitoring

### DISCUSSION THREAD

**Discussion by ghwaldo** [arendst/Tasmota#15344] [Q&A] [answered]
Hi there,

i have a plug with power monitoring. Can somebody tell me if the values EnergyTody, EnergyTotal are summarized by Tasmota or by hardware chip ?
The problem: every 5 seconds i get a value from the plug about 60 to max. 70W.
But after half of an hour it says EnergyToday 0.074 kwh.

The expected value would be 0.07 kwh in one hour, but not in 30 minutes.

Can anyone explaint to me !?

**Comment by sfromis** [comment, marked as answer]
You did not mention what EnergyToday was at the start of your half hour period.

How it works depends on what your hardware is. For most chips, Tasmota is doing the totaling. To ensure good processing of the values, you should do a calibration of the energy monitoring before assuming it to work properly.

**Comment by ghwaldo** [comment]
Sorry for not mentioning. To get rid of erors i resettet enrgyToday at 12:52. And at 13:52 it reported 0.1xx kwh. 
the actual consumption is displayed as graph, but it never reaches 70w in the last hour. avg was about 63w during last hour.

that is why i wondered how the energytotal is summarized. i have calibrated with a 60w light bulb (not LED of course).

if the calibration was bad i would expect other cinsumption values, but even then over time of an hour it is just a simple multiply.

![Unbenannt](https://user-images.githubusercontent.com/44200786/162737561-f2f45820-f2a4-4372-9c80-f92602c5bff9.png)

Attached the graph of consumption and GPIO (incl. PM Chip)
![Unbenannt2](https://user-images.githubusercontent.com/44200786/162740898-1d336bff-8485-4e5d-8fc7-da99d5cff19f.png)

**Comment by arendst** [comment]
What is the used (verified from blakadder's template list) template for this device?

It might well be that your implementation is not using HLW8012 but BL0937 which uses different CF pulse counts resulting in what you see.

**Comment by ghwaldo** [comment]
Thank you, i tried. Seems better now.
One question though: when nothing is plugged in, but plug is switched on,  it shows about 40W ReactivePower, which goes to zero, when the load gets on. Normal ? Hardware problem ?

Thanks very much so far.
